### PR TITLE
Fix Bug 1444518 - Ignore bookmarks added and removed during an import or restore

### PR DESCRIPTION
**Pull request by kitcambridge**
I think I did this right, but please let me know if there's anything that needs changing! 😄

### CODE REVIEW

**Comment by sarracini** [comment]
Thanks for doing this Kit! Maybe just add a test in PlacesFeed.test.js as well, similar to this: https://github.com/mozilla/activity-stream/blob/master/system-addon/test/unit/lib/PlacesFeed.test.js#L387

File: system-addon/lib/HighlightsFeed.jsm [hunk 1]
```diff
@@ -55,11 +59,18 @@ this.HighlightsFeed = class HighlightsFeed {
...
 
   observe(subject, topic, data) {
-    // When we receive a notification that a sync has happened for bookmarks, refresh highlights
-    if (topic === SYNC_BOOKMARKS_FINISHED_EVENT && data === "bookmarks") {
+    // When we receive a notification that a sync has happened for bookmarks,
+    // or Places finished importing or restoring bookmarks, refresh highlights
+    const manyBookmarksChanged =
+      (topic === SYNC_BOOKMARKS_FINISHED_EVENT && data === "bookmarks") ||
+      topic === BOOKMARKS_RESTORE_SUCCESS_EVENT ||
+      topic === BOOKMARKS_RESTORE_FAILED_EVENT;
```
Comment (sarracini) [inline system-addon/lib/HighlightsFeed.jsm]: I'm not sure we actually _do_ want to do a refresh if an restore has failed. Maybe you can give some more context about when this event is fired, and why we would want to fetch new highlights in this case?

Comment (kitcambridge) [reply system-addon/lib/HighlightsFeed.jsm]: That makes sense. If a restore fails, it'll erase all your existing bookmarks, but not add any new ones...so I guess the only issue is that existing highlights will be stale. But, Places tries its best to validate and fix up malformed backups before erasing everything, so do you think it's worth removing `BOOKMARKS_RESTORE_FAILED_EVENT`?

Comment (sarracini) [reply system-addon/lib/HighlightsFeed.jsm]: Does the event still fire even if Places manages to fix things up and not erase everything?

Comment (kitcambridge) [reply system-addon/lib/HighlightsFeed.jsm]: Nope, it'll be `SUCCESS` in that case.

Comment (sarracini) [reply system-addon/lib/HighlightsFeed.jsm]: Ok then I think it's ok to leave the failed even. If it really truly does fail and the bookmarks are all gone then we should reflect that in Activity Stream

**Comment by as-pine-proxy** [comment]
[Treeherder: 4a0ce47e822703a4aa3939f26ea792cc9833f959](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=4a0ce47e822703a4aa3939f26ea792cc9833f959)

**Pull request merged**